routers/repo/repo.go

```diff
@@ -142,7 +147,11 @@ func Create(ctx *context.Context) {
 		}
 	}
 
-	ctx.HTML(200, tplCreate)
+	if !ctx.User.CanCreateRepo() {
```

Comment:
Actually, `CanCreateRepo` is meant to check more things than it's actually checking. This `if` here ties the failure result to this specific error. Could you please add a note to `CanCreateRepo` like _"NOTE: functions calling this asume a failure due to repository count limit; if new checks are added, those functions should be revised"_?
Just so you don't need to create a specific error object and check for it, which is currently unnecessary.